internal/services/authorization/role_management_policy_resource.go

```diff
@@ -0,0 +1,1097 @@
+		notificationLevel = v
+	}
+	if v, ok := (*rule)["isDefaultRecipientsEnabled"].(bool); ok {
+		defaultRecipients = v
+	}
+	if v, ok := (*rule)["notificationRecipients"].([]interface{}); ok {
+		additionalRecipients = make([]string, len(v))
+		for i, r := range v {
+			additionalRecipients[i] = r.(string)
+		}
+	}
+	return &RoleManagementPolicyNotificationSettings{
+		NotificationLevel:    notificationLevel,
+		DefaultRecipients:    defaultRecipients,
+		AdditionalRecipients: additionalRecipients,
+	}
+}
+
+func getScopedPolicyId(ctx context.Context, metadata sdk.ResourceMetaData, id *parse.RoleManagementPolicyId, roleDefinitionId string) (*rolemanagementpolicies.ScopedRoleManagementPolicyId, error) {
+	if id.Name != "" {
+		// Check if the current ID is still valid
+		scopedId := pointer.To(rolemanagementpolicies.NewScopedRoleManagementPolicyID(id.Scope, id.Name))
+
+		resp, err := metadata.Client.Authorization.RoleManagementPoliciesClient.Get(ctx, *scopedId)
+		switch {
+		case err != nil && !response.WasNotFound(resp.HttpResponse):
+			return nil, fmt.Errorf("failed to get existing Role Management Policy %s. %+v", scopedId.ID(), err)
+		case resp.Model != nil:
+			return scopedId, nil
+		}
+	}
+
+	// If it wasn't found (or no resource ID was provided), we need to search for the policy ID
+	resp, err := metadata.Client.Authorization.RoleManagementPolicyAssignmentsClient.ListForScope(ctx, commonids.NewScopeID(id.Scope))
+	if err != nil {
+		return nil, fmt.Errorf("failed to list Role Management Policy Assignments for scope %s. %+v", id.Scope, err)
+	}
+
+	for _, assignment := range *resp.Model {
+		if *assignment.Properties.RoleDefinitionId == roleDefinitionId {
+			parts := strings.Split(*assignment.Properties.PolicyId, "/providers/Microsoft.Authorization/roleManagementPolicies/")
+			scopedId := pointer.To(rolemanagementpolicies.NewScopedRoleManagementPolicyID(parts[0], parts[1]))
+			return scopedId, nil
+		}
+	}
+
+	return nil, fmt.Errorf("could not find Role Management Policy for scope %s and role definition %s", id.Scope, roleDefinitionId)
+}
+
+func notificationRuleSchema() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"admin_notifications": {
+			Description: "Admin notification settings",
+			Type:        pluginsdk.TypeList,
+			Optional:    true,
+			Computed:    true,
+			MaxItems:    1,
+			Elem: &pluginsdk.Resource{
+				Schema: notificationSettingsSchema(),
+			},
+		},
+		"approver_notifications": {
+			Description: "Approver notification settings",
+			Type:        pluginsdk.TypeList,
+			Optional:    true,
+			Computed:    true,
+			MaxItems:    1,
+			Elem: &pluginsdk.Resource{
+				Schema: notificationSettingsSchema(),
+			},
+		},
+		"assignee_notifications": {
+			Description: "Assignee notification settings",
+			Type:        pluginsdk.TypeList,
+			Optional:    true,
+			Computed:    true,
+			MaxItems:    1,
+			Elem: &pluginsdk.Resource{
+				Schema: notificationSettingsSchema(),
+			},
+		},
```

Comment:
Can we space these out per the schema above for readability? :)

```suggestion
		"admin_notifications": {
			Description: "Admin notification settings",
			Type:        pluginsdk.TypeList,
			Optional:    true,
			Computed:    true,
			MaxItems:    1,
			Elem: &pluginsdk.Resource{
				Schema: notificationSettingsSchema(),
			},
		},

		"approver_notifications": {
			Description: "Approver notification settings",
			Type:        pluginsdk.TypeList,
			Optional:    true,
			Computed:    true,
			MaxItems:    1,
			Elem: &pluginsdk.Resource{
				Schema: notificationSettingsSchema(),
			},
		},

		"assignee_notifications": {
			Description: "Assignee notification settings",
			Type:        pluginsdk.TypeList,
			Optional:    true,
			Computed:    true,
			MaxItems:    1,
			Elem: &pluginsdk.Resource{
				Schema: notificationSettingsSchema(),
			},
		},
```